src/main/java/org/folio/circulation/domain/notice/schedule/ScheduledNotice.java

```diff
@@ -3,44 +3,43 @@
 import org.joda.time.DateTime;
 
 public class ScheduledNotice {
-
   private final String id;
-  private final String loanId;
-  private final String requestId;
-  private final String recipientUserId;
   private final TriggeringEvent triggeringEvent;
   private final DateTime nextRunTime;
   private final ScheduledNoticeConfig configuration;
+  private final ReferencedIds referencedIds;
+
   public ScheduledNotice(String id,
-                         String loanId,
-                         String requestId,
-                         String recipientUserId,
-                         TriggeringEvent triggeringEvent,
-                         DateTime nextRunTime,
-                         ScheduledNoticeConfig configuration) {
+    TriggeringEvent triggeringEvent,
+    DateTime nextRunTime,
+    ScheduledNoticeConfig configuration,
+    ReferencedIds referencedIds) {
+
     this.id = id;
-    this.loanId = loanId;
-    this.requestId = requestId;
-    this.recipientUserId = recipientUserId;
     this.triggeringEvent = triggeringEvent;
     this.nextRunTime = nextRunTime;
     this.configuration = configuration;
+    this.referencedIds = referencedIds;
   }
 
   public String getId() {
     return id;
   }
 
   public String getLoanId() {
-    return loanId;
+    return referencedIds.loanId;
```

Comment:
Is this NPE-safe? This class has a public constructor.

Reply:
Removed this nested class altogether.